### PR TITLE
Added new event handler subscriptions

### PR DESCRIPTION
**Pull request by sachiniWettasinghe**
### **Proposed changes in this pull request**

- Subscriptions of new event handlers were added to the identity-event.properties file.
- New event constants were added to IdentityEventConstants.
- EventListener configurations of DASLoginDataPublisherImpl and DASSessionDataPublisherImpl were removed from the identity.xml file.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthenticationDataPublisher.java [hunk 1]
```diff
@@ -22,9 +22,15 @@
...
 
-import javax.servlet.http.HttpServletRequest;
 import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
 
+/**
+ * The data publishing changed to publish data through event handling. {@link AuthenticationDataPublisher} interface is
+ * not to be used to register data publishing services from this release.
+ * {@link org.wso2.carbon.identity.event.event.Event} can be used to create a event and use
+ * {@link org.wso2.carbon.identity.event.handler.IdentityEventServiceImpl} \} to handle the event.
```
Comment (Abilashini) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthenticationDataPublisher.java]: Why do we have an additional closing curly bracket ("\}") here?

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java [hunk 2]
```diff
@@ -86,6 +86,21 @@ private Event(){}
...
+     */
+    public enum EventName {
+
+        AUTHENTICATION_STEP_SUCCESS,
+        AUTHENTICATION_STEP_FAILURE,
+        AUTHENTICATION_SUCCESS,
+        AUTHENTICATION_FAILURE,
+        SESSION_CREATE,
+        SESSION_UPDATE,
+        SESSION_TERMINATE,
```
Comment (Abilashini) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java]: Shall we remove the comma after the last element?

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java [hunk 3]
```diff
@@ -86,6 +86,21 @@ private Event(){}
...
+
```
Comment (Abilashini) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java]: Please remove the unnecessary black line

Comment (sachiniWettasinghe) [reply components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java]: Fixed with 15abde06a59a32d51f777778b9551910515589ca

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthenticationDataPublisher.java (hunk 1 above)
Comment (sachiniWettasinghe) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthenticationDataPublisher.java]: Fixed with 15abde06a59a32d51f777778b9551910515589ca

**Pull request merged**